R/load.R

```diff
@@ -876,12 +882,19 @@ renv_load_report_synchronized <- function(project = NULL, lockfile = NULL) {
   if (length(intersect(lockpkgs, libpkgs)) == 0 && length(lockpkgs) > 0L) {
 
     caution("- None of the packages recorded in the lockfile are currently installed.")
-    if (renv_rstudio_autoloading()) {
-      caution("- Use `renv::restore()` to restore the project library.")
+    autoloading <- getOption("renv.autoloader.running", default = FALSE)
+    if (autoloading) {
+      if (requireNamespace("cli", quietly = TRUE)) {
+        cli::cli_bullets(c(
+          "*" = "Use {.run renv::restore()} to restore the project library."
+        ))
+      } else {
+        caution("- Use `renv::restore()` to restore the project library.")
+      }
```

Comment:
Main change: Switch condition from `renv_rstudio_autoloading()` to just checking if we're autoloading. Because I think we're all in agreement that we should only message in this case, regardless of the frontend.

The second change, which I think we'll have to abandon, is kind of interesting and worth discussing first: try to emit the advice to run `renv::restore()` as a clickable hyperlink.

It turns out you can never hit the `cli::cli_bullet()` code above. Why not? Because the check for cli happens with respect to the current libpaths, meaning with respect to the current renv project.

And if cli is there then it's impossible that `length(intersect(lockpkgs, libpkgs)) == 0`.

So, since there's no ability to use packages for the UI that aren't present in the project-specific library, this particular effort is pointless.

There are, no doubt, other places where we *could* actually emit runnable hyperlinks, but I think the cli-using smarts would probably be better baked into `caution()` and friends.